lib/MySQL_PreparedStatement.cpp

```
@@ -50,14 +51,30 @@ static uint64_t stmt_compute_hash(char *user,
 	l += strlen(_COMPUTE_HASH_DEL2_);
 
 	// write query
-	memcpy(buf + l, query, query_length);
-	l += query_length;
+	if (strip_query_comments) {
+		char *q = mysql_query_strip_comments((char *)query, query_length, false);
+		size_t q_len = strlen(q);
+		memcpy(buf + l, q, q_len);
```

Cppcheck (2.19.0)

[error] If memory allocation fails

(nullPointerArithmeticOutOfMemory)